src/main/resources/data/computercraft/lua/rom/apis/rednet.lua

```diff
@@ -414,14 +414,15 @@ function run()
                 end
             end
 
-        elseif sEvent == "timer" then
+        elseif sEvent == "timer" and p1 == nClearTimer then
             -- Got a timer event, use it to clear the event queue
-            local nTimer = p1
-            local nMessage = tReceivedMessageTimeouts[nTimer]
-            if nMessage then
-                tReceivedMessageTimeouts[nTimer] = nil
-                tReceivedMessages[nMessage] = nil
+            nClearTimer = nil
+            local nNow, bFound = os.epoch("utc"), nil
```

Comment:
Maybe "hasMore" rather than "found"? Not clear what it's found here. Similarly, would be inclined to call `k`/`v` something more intuitive (a break from tradition, I know!) as in this case their meaning is a little subtle.